File: problems/2570-merge-two-2d-arrays-by-summing-values/solution.java
import java.util.ArrayList;
import java.util.List;
import java.util.Map;
import java.util.TreeMap;

class Solution {
    public int[][] mergeArrays(int[][] nums1, int[][] nums2) {
        Map<Integer, Integer> map = new TreeMap<>();
        for (int[] num : nums1) {
            map.put(num[0], num[1]);
        }
        for (int[] num : nums2) {
            map.put(num[0], map.getOrDefault(num[0], 0) + num[1]);
        }

        List<int[]> result = new ArrayList<>();
        for (Map.Entry<Integer, Integer> entry : map.entrySet()) {
            result.add(new int[]{entry.getKey(), entry.getValue()});
        }

        return result.toArray(new int[result.size()][]);
    }
}

File: problems/2570-merge-two-2d-arrays-by-summing-values/solution.py
class Solution:
    def mergeArrays(self, nums1: List[List[int]], nums2: List[List[int]]) -> List[List[int]]:
        merged = {}
        for id, val in nums1:
            merged[id] = val
        for id, val in nums2:
            if id in merged:
                merged[id] += val
            else:
                merged[id] = val
        result = []
        for id in sorted(merged.keys()):
            result.append([id, merged[id]])
        return result